airbyte-integrations/connectors/source-salesforce/source_salesforce/spec.yaml

```diff
@@ -44,18 +44,25 @@ connectionSpecification:
     start_date:
       title: Start Date
       description: >-
-        Enter the date in the YYYY-MM-DD format. Airbyte will replicate the data added on and after this date. If this field is blank, Airbyte will replicate the data for last two years.
+        Enter the date (or date-time) in the YYYY-MM-DD or YYYY-MM-DDTHH:mm:ssZ format. Airbyte will replicate the data added on and after this date. If this field is blank, Airbyte will replicate the data for last two years.
       type: string
       pattern: >-
         ^([0-9]{4}-[0-9]{2}-[0-9]{2}(T[0-9]{2}:[0-9]{2}:[0-9]{2}Z)?)$
+      pattern_descriptor: "YYYY-MM-DD or YYYY-MM-DDTHH:mm:ssZ"
       examples:
         - "2021-07-25"
         - "2021-07-25T00:00:00Z"
       format: date-time
       order: 5
+    force_use_bulk_api:
```

Comment:
We shouldn't expose this option. API type is an implementation detail

Reply:
Do you want to use another naming?

Reply:
For our current [OC issue](https://github.com/airbytehq/oncall/issues/2409) customer has unsupported fields in `Accounts` stream and millions of records to download. How should we handle this case without explicit exposing Salesforce API type ?

Reply:
> Can't we detect that based on the column type?

Don't really understand what you mean.

We choose which API type to use base on some conditions (BULK is preferred, but if there are some exceptions, like stream is not queryable or some unsupported columns are presented in schema, then we'll use REST API).
With this feature customers will be able to use BULK API instead of REST API, even if we have some unsupported columns (we'll just skip them from query).

Reply:
> customer has unsupported fields in Accounts stream
Is it possible for us to discover if the fields are unsupported upfront and select the API type based on that? (don't we already do that if e.g the column is binary)

Reply:
It does work so already (see `def _get_api_type`). But for [OC customer](https://github.com/airbytehq/oncall/issues/2409) this method will select the REST API as default one (because he has some `objects` in `schema`. Downloading huge amount of data (90 millions of records) through REST API is not optimal (will lead to timeouts, exceeding the API limits). This feature will give the customer an opportunity to force our connector to use BULK API.

Reply:
OK i see your point, that seems like a valid problem to solve. Ideally we make this selection based on the volume of data we're about to sync. Is it possible to detect this and choose the correct API type accordingly?

Reply:
So, we could implement next algorithm:
Before class (stream) instantiation make a `Select count(*)` query to check for data amount in slice. Slicing is still required due to potential timeouts. Lets say, If [number of records > 2k](https://help.salesforce.com/s/articleView?id=sf.integrate_what_is_api.htm&type=5#:~:text=When%20to%20Use%20Bulk%20API%202.0) , we'll force to use BULK API.
But I have some concerns about this:
1. Speed: Count function in salesforce is not as fast as in DB.
2. API Limitations for bulk|rest operations: this extra step will double requests for every sync.

Or, we could leave this feature, and add link to salesforce docs: [when to use bulk api](https://help.salesforce.com/s/articleView?id=sf.integrate_what_is_api.htm&type=5#:~:text=When%20to%20Use%20Bulk%20API%202.0)

Reply:
Added. https://github.com/airbytehq/airbyte/issues/29301